File: snippets/fsharp/VS_Snippets_CLR_System/system.Int32.TryParse/fs/TryParse1.fs
module TryParse1

// <Snippet1>
open System

let values = 
   [ null; "160519"; "9432.0"; "16,667"
     "   -322   "; "+4302"; "(100);"; "01FA" ]
for value in values do
    match Int32.TryParse value with
    | true, number -> 
        printfn $"Converted '{value}' to {number}."
    | _ -> 
        printfn $"""Attempted conversion of '{if isNull value then "<null>" else value}' failed."""
         
// The example displays the following output:
//       Attempted conversion of '<null>' failed.
//       Converted '160519' to 160519.
//       Attempted conversion of '9432.0' failed.
//       Attempted conversion of '16,667' failed.
//       Converted '   -322   ' to -322.
//       Converted '+4302' to 4302.
//       Attempted conversion of '(100);' failed.
//       Attempted conversion of '01FA' failed.
// </Snippet1>

File: snippets/csharp/System/Int64/TryParse/TryParse1.cs
// <Snippet1>
using System;

public class Int64TryParseExample1
{
    public static void Run()
    {
        TryToParse(null);
        TryToParse("160519");
        TryToParse("9432.0");
        TryToParse("16,667");
        TryToParse("   -322   ");
        TryToParse("+4302");
        TryToParse("(100);");
        TryToParse("01FA");
    }

    private static void TryToParse(string value)
    {
        bool success = long.TryParse(value, out long number);
        if (success)
        {
            Console.WriteLine($"Converted '{value}' to {number}.");
        }
        else
        {
            if (value == null) value = "";
            Console.WriteLine($"Attempted conversion of '{value}' failed.");
        }
    }
}
// The example displays the following output to the console:
//       Attempted conversion of '' failed.
//       Converted '160519' to 160519.
//       Attempted conversion of '9432.0' failed.
//       Attempted conversion of '16,667' failed.
//       Converted '   -322   ' to -322.
//       Converted '+4302' to 4302.
//       Attempted conversion of '(100);' failed.
//       Attempted conversion of '01FA' failed.
// </Snippet1>
